### PR TITLE
Rebuilt Threads Web UI using the app router of NextJS. closes #107

### LINKED ISSUE #107: Remake threads-web-ui using the Next JS app router.
- The current `threads-web-ui` is built using the the old version of NextJS (using the `/pages` router). 
- It is advisable to migrate it to the newest version that makes use of the stable `/app` router. 
PS: I can work on it and send a PR.

### PR DESCRIPTION
**Pull request by sudarshanmg**
- Please have a look at the code
- closes #107 
- Still a beginner

### CODE REVIEW

File: threads-web-ui/.gitignore [hunk 1]
```diff
@@ -25,10 +25,11 @@ yarn-debug.log*
...
-.env.test.local
-.env.production.local
+.env*.local
 
 # vercel
 .vercel
+
+# typescript
+*.tsbuildinfo
+next-env.d.ts
```
Comment (junhoyeo) [inline threads-web-ui/.gitignore:35]: Why are you ignoring this file from version control?

Comment (junhoyeo) [inline threads-web-ui/.gitignore:35]:
```suggestion
next-env.d.ts
```
Why are you ignoring this file from version control?

File: threads-web-ui/app/[username]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  if (!userID) {
+    return <div>No user with the given username</div>;
+  }
+
+  const values: [string, string] = [username, userID];
+
+  const [userProfile, userThreads] = await Promise.all([
+    threadsAPI.getUserProfile(...values),
+    threadsAPI.getUserProfileThreads(...values),
+  ]);
```
Comment (junhoyeo) [inline threads-web-ui/app/[username]/page.tsx:28]:
```suggestion
  const params: [string, string] = [username, userID];
  const [userProfile, userThreads] = await Promise.all([
    threadsAPI.getUserProfile(...params),
    threadsAPI.getUserProfileThreads(...params),
  ]);
```
`values` is way to general

File: threads-web-ui/app/globals.css [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+      html,
+      body {
+        background-color: #101010;
+      }
+
+      body {
+        color: slategray;
+      }
+
```
Comment (junhoyeo) [inline threads-web-ui/app/globals.css:108]: can you format this file? maybe `yarn format` will handle this for u

File: threads-web-ui/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { GlobalStyle } from "@/components/GlobalStyle";
+import "./globals.css";
+import type { Metadata } from "next";
+import { Inter } from "next/font/google";
```
Comment (junhoyeo) [inline threads-web-ui/app/layout.tsx:4]: `yarn format` should order imports

File: threads-web-ui/next.config.js [hunk 5]
```diff
@@ -1,18 +1,12 @@
...
-      {
-        protocol: 'https',
-        hostname: '**.fbcdn.net',
+        hostname: "scontent.cdninstagram.com",
       },
     ],
   },
 };
+
+module.exports = nextConfig;
```
Comment (junhoyeo) [inline threads-web-ui/next.config.js:12]: Can we roll back this? Some CDN hosts are different

File: threads-web-ui/package.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
   "name": "threads-web-ui",
   "version": "1.0.0",
   "private": true,
-  "repository": "https://github.com/junhoyeo/threads-web-ui",
+  "repository": "",
```
Comment (junhoyeo) [inline threads-web-ui/package.json:5]: can we revert this bro?

**Review by junhoyeo — CHANGES_REQUESTED** [review]
Thank you so much @sudarshanmg 👍👍 Love if you check reviews!

**Comment by sudarshanmg** [comment]
On it! Thanks

File: threads-web-ui/.gitignore (hunk 1 above)
Comment (sudarshanmg) [reply threads-web-ui/.gitignore:35]: The `next-env.d.ts` file is not required for version control purposes. It is a TypeScript declaration file specific to Next.js projects. This file is automatically generated by Next.js when you run the development server (next dev) or build your application (next build).
It is recommended that each developer generates their own `next-env.d.ts` file to ensure compatibility with their local development environment and TypeScript setup.

Comment (sudarshanmg) [reply threads-web-ui/.gitignore:35]: I see :eyes: 
Okay! I will remove it from the `.gitignore` file. :smile:

**Comment by sudarshanmg** [comment]
Please have a look at it now.
I made all the changes you requested.

File: threads-web-ui/app/[username]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  if (!userID) {
+    return <div>No user with the given username</div>;
+  }
+
+  const userInfo: [string, string] = [username, userID];
+
+  const [userProfile, userThreads] = await Promise.all([
+    threadsAPI.getUserProfile(...userInfo),
+    threadsAPI.getUserProfileThreads(...userInfo),
+  ]);
```
Comment (junhoyeo) [inline threads-web-ui/app/[username]/page.tsx:24]: - Super nit I still think it should be `params`
- But this is unnecessary anymore, I changed the interface in #121, so we can just change `userID` now. 😁 
- So I'll just pass this. 👍

File: threads-web-ui/next.config.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ const nextConfig = {
     remotePatterns: [
       {
         protocol: 'https',
-        hostname: '**.cdninstagram.com',
+        hostname: '**.cdninstagram.com, **.fbcdn.net',
```
Comment (junhoyeo) [inline threads-web-ui/next.config.js:7]: I dont think this will work

File: threads-web-ui/package.json (hunk 6 above)
Comment (junhoyeo) [reply threads-web-ui/package.json:5]: can we have a bump here

**Comment by junhoyeo** [comment]
I'll just squash-merge this and open another PR that we can exchange thoughts! Thank you so much for this work @sudarshanmg 👍👍

File: threads-web-ui/next.config.js (hunk 8 above)
Comment (junhoyeo) [reply threads-web-ui/next.config.js:7]: image
See

**Pull request merged**